github/github.go

```diff
@@ -399,7 +399,7 @@ func parseRate(r *http.Response) Rate {
 // The provided ctx must be non-nil. If it is canceled or times out,
 // ctx.Err() will be returned.
 func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Response, error) {
-	ctx, req = withContext(ctx, req)
+	req = withContext(ctx, req)
```

Comment:
This is one of those rare times where I feel it might be worth breaking the "context is always the first parameter" pattern.

Here, `withContext` is basically a method-as-a-function that adds `ctx` to `req`. I think seeing it like this:

```Go
req = withContext(req, ctx)
```

Might read better...

Well, after writing it up and looking at it for a few minutes, I'm not so sure it looks better. It seems about about the same. 😑